agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/DefaultClassNamesProvider.java

```diff
@@ -0,0 +1,215 @@
+
+    private final HashMap<String, ClassInstrumentationData> classesToInstrument = new HashMap<String, ClassInstrumentationData>();
+
+    private boolean builtInEnabled = true;
+
+    public DefaultClassNamesProvider() {
+        populateForbiddenClasses();
+    }
+
+    @Override
+    public void setConfiguration(AgentConfiguration agentConfiguration) {
+        this.agentConfiguration = agentConfiguration;
+        addConfigurationData();
+
+        if (builtInEnabled) {
+            InternalAgentLogger.INSTANCE.trace("Adding built-in instrumentation");
+
+            populateSqlClasses();
+            populateHttpClasses();
+            addHibernate();
+        }
+    }
+
+    @Override
+    public boolean isSqlClass(String className) {
+        return sqlClasses.contains(className);
+    }
+
+    @Override
+    public boolean isHttpClass(String className) {
+        return httpClasses.contains(className);
+    }
+
+    /**
+     * Gets the {@link ClassInstrumentationData} that is associated
+     * with the argument 'className', and removes that entry from the container once this is found
+     * @param className The class name to search for
+     * @return The {@link ClassInstrumentationData}
+     */
+    @Override
+    public ClassInstrumentationData getAndRemove(String className) {
+        return classesToInstrument.remove(className);
+    }
+
+    private void addHibernate() {
+        HashSet<String> methodNamesOnly = new HashSet<String>();
+        methodNamesOnly.add("delete");
+        methodNamesOnly.add("execute");
+        methodNamesOnly.add("executeNativeUpdate");
+        methodNamesOnly.add("executeUpdate");
+        methodNamesOnly.add("find");
+        methodNamesOnly.add("get");
+        methodNamesOnly.add("save");
+        methodNamesOnly.add("list");
+        methodNamesOnly.add("load");
+        methodNamesOnly.add("saveOrUpdate");
+        methodNamesOnly.add("update");
+
+        addToClasses("org/hibernate/impl/SessionImpl", InstrumentedClassType.SQL, methodNamesOnly);
+
+        methodNamesOnly.clear();
+        methodNamesOnly.add("delete");
+        methodNamesOnly.add("get");
+        methodNamesOnly.add("insert");
+        methodNamesOnly.add("list");
+        methodNamesOnly.add("update");
+
+        addToClasses("org/hibernate/impl/StatelessSessionImpl", InstrumentedClassType.SQL, methodNamesOnly);
+    }
+
+    private void populateForbiddenClasses() {
+        forbiddenPaths.add("java/");
+        forbiddenPaths.add("java/");
+        forbiddenPaths.add("javax/");
+        forbiddenPaths.add("org/apache");
+        forbiddenPaths.add("com/microsoft/applicationinsights");
+        forbiddenPaths.add("sun/nio/");
+        forbiddenPaths.add("sun/rmi/");
+        forbiddenPaths.add("com/sun/jmx/");
+        forbiddenPaths.add("sun/net/www/http/KeepAlive");
+        forbiddenPaths.add("com.google");
+    }
+
+    private void populateSqlClasses() {
```

Comment:
For better modularity I think it would be better to have SQL and Http in separate classes